core/App/ehsm_provider.cpp

```diff
@@ -149,91 +149,53 @@ void Finalize()
     sgx_destroy_enclave(g_enclave_id);
 }
 
-static uint32_t sgx_calc_gcm_data_size(const uint32_t aad_size, const uint32_t plaintext_size)
+static uint32_t sgx_get_rsa_cipher_len(const uint32_t keyspec)
 {
-    if (aad_size > UINT32_MAX - sizeof(aes_gcm_key_data_t))
-        return UINT32_MAX;
-
-    if (plaintext_size > UINT32_MAX - sizeof(aes_gcm_key_data_t))
-        return UINT32_MAX;
-
-    if (aad_size > UINT32_MAX - plaintext_size)
-        return UINT32_MAX;
-
-    if (sizeof(aes_gcm_key_data_t) > UINT32_MAX - plaintext_size - aad_size)
-        return UINT32_MAX;
+    switch (keyspec) {
+        case EH_RSA_2048:
+            return RSA_OAEP_2048_CIPHER_LENGTH;
+        case EH_RSA_3072:
+            return RSA_OAEP_3072_CIPHER_LENGTH;
+        default:
+            return 0;
+    }
+}
 
-    return (aad_size + plaintext_size + sizeof(aes_gcm_key_data_t));
+static uint32_t sgx_get_rsa_padding_size(uint8_t padding_mode)
+{
+    switch (padding_mode) {
+        case RSA_PKCS1_OAEP_PADDING:
+            return 41;
+        case RSA_PKCS1_PADDING:
+            return 11;
```

Comment:
where did you get these value?

Reply:
They are the space that must be reserved for the padding mode, the rsa encryption plaintext length must be keysize-padding size.
I also look if there are the macros defined by sgx-ssl